sherpa-ncnn/csrc/sherpa-ncnn.cc

```diff
@@ -48,38 +45,38 @@ You can download pre-trained models from the following repository:
 
     return 0;
   }
-  sherpa_ncnn::ModelConfig config;
-
-  config.tokens = argv[1];
-  config.encoder_param = argv[2];
-  config.encoder_bin = argv[3];
-  config.decoder_param = argv[4];
-  config.decoder_bin = argv[5];
-  config.joiner_param = argv[6];
-  config.joiner_bin = argv[7];
-
-  std::string wav_filename = argv[8];
-
-  int32_t num_threads = 4;
+  sherpa_ncnn::ModelConfig model_conf;
+  model_conf.tokens = argv[1];
+  model_conf.encoder_param = argv[2];
+  model_conf.encoder_bin = argv[3];
+  model_conf.decoder_param = argv[4];
+  model_conf.decoder_bin = argv[5];
+  model_conf.joiner_param = argv[6];
+  model_conf.joiner_bin = argv[7];
+  int num_threads = 4;
   if (argc == 10) {
     num_threads = atoi(argv[9]);
```

Comment:
Could you add a check to ensure that `num_threads > 0`?
If a user forgets to provide `num_threads` but provide decoding method, it will treat decoding method as num_threads.